src/main/java/com/michelin/ns4kafka/service/executor/UserAsyncExecutor.java

```diff
@@ -200,6 +233,66 @@ public String resetPassword(String user) {
             return password;
         }
 
+        @Override
+        public void setPassword(String user, String password) {
+            UserScramCredentialUpsertion update = new UserScramCredentialUpsertion(user, info, password);
+            try {
+                admin.alterUserScramCredentials(List.of(update)).all().get(10, TimeUnit.SECONDS);
+                log.info("Success setting password for user {}", user);
+            } catch (InterruptedException e) {
+                log.error("Error", e);
+                Thread.currentThread().interrupt();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public boolean checkPassword(String user, String password, Properties config) {
+            Properties props = new Properties();
+            props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, config.get("bootstrap.servers"));
```

Comment:
Using an `AdminClient` to verify the connexion may be more elegant than instantiating a dedicated producer, let me sleep on that